test/integration/consul-container/libs/service/helpers.go

```diff
@@ -35,7 +35,7 @@ type ServiceOpts struct {
 }
 
 // createAndRegisterStaticServerAndSidecar register the services and launch static-server containers
-func createAndRegisterStaticServerAndSidecar(node libcluster.Agent, grpcPort int, svc *api.AgentServiceRegistration) (Service, Service, error) {
+func createAndRegisterStaticServerAndSidecar(node libcluster.Agent, grpcPort int, svc *api.AgentServiceRegistration, containerArgs ...string) (Service, Service, error) {
```

Comment:
looks like containerArgs is added, but it's used nowhere (I might be missing something.)

Reply:
it's used in creating the services that the api gateway is going to route to in order to disambiguate between which service is getting hit when issuing an HTTP request:

```go
	serviceTwo := createService(t, cluster, &libservice.ServiceOpts{
		Name:     "service2",
		ID:       "service2",
		HTTPPort: 8081,
		GRPCPort: 8082,
	}, []string{
		"-echo-server-default-params", fmt.Sprintf("status=%d", service2ResponseCode),
	},
```

and then later on:

```go
	checkRoute(t, ip, gatewayPort, "v2", map[string]string{
		"Host": "test.foo",
		"x-v2": "v2",
	}, checkOptions{statusCode: service2ResponseCode, testName: "service2 header and path"})
```

basically setting the status code the server is going to return and then verifying that it does indeed return that so we know we're hitting that service and not some other.

Reply:
Thanks for clarifying this.